src/mame/namco/toypop.cpp

```diff
@@ -79,61 +85,59 @@ class namcos16_state : public driver_device
 	required_device<palette_device> m_palette;
 	required_device<gfxdecode_device> m_gfxdecode;
 
-	required_shared_ptr<uint8_t> m_master_workram;
-	required_shared_ptr<uint8_t> m_slave_sharedram;
-	required_shared_ptr<uint16_t> m_bgvram;
-	required_shared_ptr<uint8_t> m_fgvram;
-	required_shared_ptr<uint8_t> m_fgattr;
+	// memory pointers
+	required_shared_ptr<u8> m_spriteram;
+	required_shared_ptr<u8> m_sharedram;
+	required_shared_ptr<u16> m_bgvram;
+	required_shared_ptr<u8> m_txvram;
+
+	required_ioport_array<2> m_io_dsw;
+
+	// video-related
+	tilemap_t *m_tx_tilemap;
+	u8 m_pal_bank = 0;
+
+	// misc
+	bool m_main_irq_enable = false;
+	bool m_sub_irq_enable = false;
+
+	TIMER_DEVICE_CALLBACK_MEMBER(main_scanline);
+	void vblank_irq(int state);
+
+	u8 irq_enable_r();
+	void irq_disable_w(u8 data);
+	void irq_ctrl_w(offs_t offset, u8 data);
+	template <unsigned Which> u8 dip_l();
+	template <unsigned Which> u8 dip_h();
+	void pal_bank_w(offs_t offset, u8 data);
+	void flip(u8 data);
+	void sub_halt_ctrl_w(offs_t offset, u8 data);
+	u8 sharedram_r(offs_t offset);
+	void sharedram_w(offs_t offset, u8 data);
+	void sub_irq_enable_w(offs_t offset, u16 data);
+	void sound_halt_ctrl_w(offs_t offset, u8 data);
+	void txvram_w(offs_t offset, u8 data);
+	u8 bg_rmw_r(offs_t offset);
+	void bg_rmw_w(offs_t offset, u8 data);
```

Comment:
... and, in cases like this causes a drift: some of these functions are video related (`bg_rmw_r/_w`), stuff in between mixes irq control logic and comms with sub CPU. It's again stuff that should be obvious by the naming context and newline separation alone.